### ISSUE TITLE
Llama Index TS can't extract nodes from a single document, created from a long text file.

### ISSUE THREAD

**Issue by preemoDez** [run-llama/LlamaIndexTS#363]
The following code:

```
const docs = await reader.loadData({ directoryPath });
const nodeParser = new SimpleNodeParser();
const nodes = nodeParser.getNodesFromDocuments(docs);
```

Does not work on the long text file. Attached is a 25 MB text file, on which this code takes forever to execute. Llama Index Python gets nodes from documents in around 20 seconds.

Could you please fix this ASAP?
[war-and-peace-loop-25mb.txt](https://github.com/run-llama/LlamaIndexTS/files/13894830/war-and-peace-loop-25mb.txt)

**Comment by logan-markewich** [comment]
Just to compare, I ran this code in python

```python
from llama_index import SimpleDirectoryReader
from llama_index.node_parser import SentenceSplitter, TokenTextSplitter

documents = SimpleDirectoryReader(
    input_files=["./war-and-peace-loop-25mb.txt"]
).load_data()

import time

start = time.time()
nodes = SentenceSplitter()(documents)
end = time.time()

print(f"Sentence splitting took {end - start} seconds")

start = time.time()
nodes = TokenTextSplitter()(documents)
end = time.time()

print(f"Token splitting took {end - start} seconds")
```

The output was

```
Sentence splitting took 34.480053424835205 seconds
Token splitting took 40.344465494155884 seconds
```

**Comment by marcusschiesser** [comment]
@preemoDez fixed by llamaindex@0.0.43

**Comment by preemoDez** [comment]
> @preemoDez fixed by llamaindex@0.0.43

Thanks!
